### PR TITLE
Implementation of share note function

### PR DESCRIPTION
**Pull request by clemlaf**
In this branch I implemented a "share note" function. This new feature required to change the Tag/User relation from Many to many to One to Many (a user has many tags), so that private tags can really be private.
On the API side I also implemented a share notebook function which is still not called by the frontend.
The Share note buttons (in the edit menu or in the navbar) shows a modal view which lists the users. The owner then can choose the permissions to give to each user in a <select> form.
share_note

### CODE REVIEW

**Comment by clemlaf** [comment]
By the way this branch answers to issue #369

**Comment by clemlaf** [comment]
I also added a new relation between version and users, and updated the note info popup and the history timeline so that the user involved in each version is shown. The point is if you shared a note with another user, then you would want to see who has been changing the notes and not only when the note was changed.
The note info then shows the creator of the note (that is to say the owner) and the user who updated the last version.
user-version

File: frontend/app/config/paperwork.php [hunk 1]
```diff
@@ -61,7 +61,7 @@
 	| If set to true, forgot password link is enabled.
 	|
 	*/
-	'forgot_password' => true,
+	'forgot_password' => false,
```
Comment (Baldrs) [inline frontend/app/config/paperwork.php]: Please revert to default value for both forgot_password and showIssueReportingLink

**Comment by Liongold** [comment]
Sorry if I'm asking too much but do you think some documentation should be done about this feature?

**Comment by Baldrs** [comment]
@Liongold he explained the problem in #369

**Comment by Liongold** [comment]
I didn't mean any problem. I meant some type of help text either in our wiki or in Paperwork itself.

**Comment by clemlaf** [comment]
I wrote a wiki page about the design choices I made in the implemention of the note sharing feature.
It's [there](https://github.com/twostairs/paperwork/wiki/Note-sharing).

**Comment by Baldrs** [comment]
Good, please revert config to defaults and we can proceed

**Comment by Liongold** [comment]
If it isn't a problem for you @Baldrs I'll proceed in merging this since the config has been reverted. Thanks a lot @clemlaf for your work.

**Pull request merged**